src/Cluster/ClusterStrategy.php

```diff
@@ -388,6 +393,24 @@ protected function getKeyFromZsetAggregationCommands(CommandInterface $command)
         return $arguments[0];
     }
 
+    /**
+     * Extracts key from SUNSUBSCRIBE command if it's given.
+     *
+     * @param  CommandInterface $command
+     * @return string
+     */
+    protected function getKeyFromSUnsubscribeCommand(CommandInterface $command): ?string
+    {
+        $arguments = $command->getArguments();
+
+        // SUNSUBSCRIBE command could be called without arguments, so it doesn't matter on each node it will be called.
+        if (empty($arguments)) {
+            return 'fake';
```

Comment:
"SUNSUBSCRIBE command could be called without arguments, so it doesn't matter on ~each~ *which* (?) node it will be called."

why `return 'fake';` and not `return null;`?

Reply:
also, why this function is here and not in the `SUNSUBSCRIBE` class?

Reply:
1. In Predis we're using pre-calculation (same algorithm that redis do) of slots to determine node that should accept command, it works well for commands that runs against specific key/keys, but if there's no key you still need to pre-calculate node so it's just random string returns to allow this action.
2. It's just because of bad design, I'm totally agree it should be a part of command object and do not violate SRP, but fixing it is beyond the scope of RESP3

Reply:
1. but that means that the node that corresponds to `'fake'` (slot 14787) will be used more than other nodes..

Reply:
For `SUNSUBSCRIBE` command only which is okay??, since it's not something you're often sends to server, but I can use randomly generated string here instead